### PR TITLE
Porting library to .NET Core

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
This is literally the last dependency in my projects that requires full framework. This PR is an attempt to support .NET Core scenaria, however no promises this will be successful :-)

Work done so far:

- [x] Migrate solution to dotnet sdk 3.0 tools.
- [x] Migrate FSharp.Formatting.Razor to nestandard2.0 by replacing `RazorEngine` with `RazorEngine.NetCore`.
- [x] Add netcoreapp3.1 targets to all unit test projects.
- [x] Make all tests green.
- [x] Use `dotnet pack` for creating nuget packages.
- [x] Convert FSharp.Formatting.CommandTool to dotnet tool.
- [x] Add a FSharp.Formatting.Razor package.

### CODE REVIEW

**Comment by thinkbeforecoding** [comment]
Yes the only part missing was the templating part in razor

**Comment by wallymathieu** [comment]
Looks like at least appveyor is happy 👍

**Comment by eiriktsarpalis** [comment]
Looks like tests are finally all green

**Comment by eiriktsarpalis** [comment]
This is now ready for review.

File: docs/tools/generate.fsx [hunk 1]
```diff
@@ -96,7 +93,7 @@ subDirectories templates
 // Copy static files and CSS + JS from F# Formatting
 let copyFiles () =
   copyRecursive files output 
-  Directory.CreateDirectory (output + "/" + "content") |> ignore
+  //Directory.CreateDirectory (output + "/" + "content") |> ignore
```
Comment (wallymathieu) [inline docs/tools/generate.fsx]: perhaps remove?

File: src/CSharpFormat/CSharpFormat.csproj [hunk 2]
```diff
@@ -1,44 +1,11 @@
...
-    <TargetFrameworks>net461;netstandard2.0</TargetFrameworks>
-    <OutputType>Library</OutputType>
-    <EnableDefaultCompileItems>false</EnableDefaultCompileItems>
-    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
-    <SolutionDir>..\..\</SolutionDir>
-    <OutputPath>$(SolutionDir)\bin</OutputPath>
-    <DocumentationFile>$(OutputPath)\$(TargetFramework)\$(AssemblyName).xml</DocumentationFile>
-    <Product>CSharpFormat</Product>
-    <Description>CSharpFormat</Description>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (wallymathieu) [inline src/CSharpFormat/CSharpFormat.csproj]: 👍

File: docs/content/metadata.fsx [hunk 3]
```diff
@@ -112,7 +113,7 @@ It needs to be set on a separate tripple-slashed line, but it could be either th
 *)
 /// [omit]
 /// Some actual comment
-module Foo = 
+module Bar =
```
Comment (wallymathieu) [inline docs/content/metadata.fsx]: 😄

File: docs/tools/generate.fsx [hunk 4]
```diff
@@ -4,18 +4,19 @@
...
 #r "FSharp.CodeFormat.dll"
 #r "FSharp.Literate.dll"
 #r "FSharp.Markdown.dll"
 #r "FSharp.MetadataFormat.dll"
 #r "FSharp.Formatting.Common.dll"
-#r "RazorEngine.dll"
+#r "Microsoft.AspNetCore.Razor.dll"
+#r "Microsoft.AspNetCore.Razor.Runtime.dll"
+#r "Microsoft.AspNetCore.Razor.Language.dll"
+#r "RazorEngine.NetCore.dll"
```
Comment (wallymathieu) [inline docs/tools/generate.fsx]: 👍

**Comment by dsyme** [comment]
So much goodness!

**Comment by matthid** [comment]
Thanks a lot for going through that hell. 
I guess that doesn't mean we can use .netstandard or .netcore dlls as input but the tools are at least now running as global tools?

**Comment by eiriktsarpalis** [comment]
It seems to be working fine with netstandard assemblies. I have added tests for that.

**Comment by dsyme** [comment]
@eiriktsarpalis You should now be admin on everything you need to update the nugets etc.  Just go for it :))

**Comment by dsyme** [comment]
After this I will try to take a stab at moving FsLab to be .NET Core only

**Comment by wallymathieu** [comment]
This is great news! 😄

**Comment by eiriktsarpalis** [comment]
I just pushed 4.0.0-alpha01 to nuget. Dogfood away!

**Comment by eiriktsarpalis** [comment]
Tested against Argu, seems to be working fine https://github.com/fsprojects/Argu/commit/a2c0319f472ead9f28ec4ba1f01b2f13f20293b4

**Comment by baronfel** [comment]
this is lovely, I might be able to update FCS's docs themselves now!

**Comment by eiriktsarpalis** [comment]
FWIW I pushed a new package, [FSharp.Formatting.Razor](https://www.nuget.org/packages/FSharp.Formatting.Razor), so everything can be referenced as vanilla nuget packages.

**Pull request merged**